adapters/repos/db/index.go

```diff
@@ -340,6 +335,12 @@ func (i *Index) determineObjectShard(id strfmt.UUID, tenantKey string, ss *shard
 			return "", fmt.Errorf("cannot find sharding state for class %q", i.Config.ClassName.String())
 		}
 	}
+	if ss.PartitioningEnabled && tenantKey == "" {
```

Comment:
For better performance we can put the validation back into `validateMultiTenancy()` 
Use `Index.PartioningEnabled` (see comment above) to avoid quering sharding state
Call `validateMultiTenancy()` first in `determineObjectShard()` before getting sharding state

Reply:
validation moved back to validateMultiTenancy() method